src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -378,6 +512,117 @@ public PreferencesViewModel(PreferenceSettings preferenceSettings, DynamoPythonS
+        /// </summary>
+        /// <param name="groupName"></param>
+        public void RemoveStyleEntry(string groupName)
+        {
+            StyleItem itemToRemove = (from item in StyleItemsList where item.GroupName.Equals(groupName) select item).FirstOrDefault();
+            StyleItemsList.Remove(itemToRemove);
+        }
+
+        /// <summary>
+        /// This method will check if the Style that is being created already exists in the Styles list
+        /// </summary>
+        /// <param name="item1"></param>
+        /// <returns></returns>
+        public bool ValidateExistingStyle(StyleItem item1)
+        {
+            return StyleItemsList.Where(x => x.GroupName.Equals(item1.GroupName)).Any();
+        }
+
+        /// <summary>
+        /// This method will remove a specific style control from the Styles list
+        /// </summary>
+        public void ResetAddStyleControl()
+        {
+            AddStyleControl.GroupName = String.Empty;
+            AddStyleControl.HexColorString = "#" + GetRandomHexStringColor();
+            IsWarningEnabled = false;
+        }
+
+        /// <summary>
+        /// This Method will generate a random color string in a Hexadecimal format
+        /// </summary>
+        /// <returns></returns>
+        public string GetRandomHexStringColor()
+        {
+            Random r = new Random();
+            Color color = Color.FromArgb(255, (byte)r.Next(), (byte)r.Next(), (byte)r.Next());
+            return ColorTranslator.ToHtml(color).Replace("#", "");
         }
     }
+
+    /// <summary>
+    /// This Class will act as a container for each of the StyleItems in the Styles list located in in the Visual Settings -> Group Styles section
+    /// </summary>
+    public class StyleItem : ViewModelBase
+    {
+        private string groupName;
+        private string hexColorString;
+
+        /// <summary>
+        /// This property will containt the Group Name thas was added by the user when creating a new Style
+        /// </summary>
+        public string GroupName
+        {
+            get { return groupName; }
+            set
+            {
+                groupName = value;
+                RaisePropertyChanged(nameof(GroupName));
+            }
+        }
+
+        /// <summary>
+        /// This property represents a color in a hexadecimal representation (with the # character at the beginning of the string)
+        /// </summary>
+        public string HexColorString
+        {
+            get { return hexColorString; }
+            set
+            {
+                hexColorString = value;
+                RaisePropertyChanged(nameof(HexColorString));
+            }
+        }
+    }
+
+    /// <summary>
+    /// This class will contain the Enum value and the corresponding description for each radio button in the Visual Settings -> Geometry Scaling section
+    /// </summary>
+    public class GeometryScalingOptions
```

Comment:
Would you double check if all the added public classed/properties need to be `public` or could be internal?

Reply:
I just added two classes StyleItem and GeometryScalingOptions, the StyleItem class need to be public since is used in public methods that are called in the code behind (If i change to internal got several errors) also the GeometryScalingOptions need to be public because there is a public property declared (GeometryScalingOptions type) and used as a binding in the PreferencesView.xaml (also is if changed to internal I got several errors).